android-auto-app/src/main/java/com/mapbox/navigation/examples/androidauto/CarAppSyncComponent.kt

```diff
@@ -144,17 +132,47 @@ class CarAppSyncComponent private constructor() : MapboxNavigationObserver {
     private val carSyncComponent = object : MapboxNavigationObserver {
         var isAttached = false
             private set
+        var carCoroutineScope: CoroutineScope? = null
         override fun onAttached(mapboxNavigation: MapboxNavigation) {
             logI(LOG_TAG, "onAttached car")
+            carCoroutineScope = MainScope()
             isAttached = true
+            carCoroutineScope?.launch {
+                MapboxCarApp.carAppState.collect { onCarAppStateUpdate(it) }
+            }
         }
 
         override fun onDetached(mapboxNavigation: MapboxNavigation) {
             isAttached = false
+            carCoroutineScope?.cancel()
+            carCoroutineScope = null
             logI(LOG_TAG, "onDetached car")
         }
     }
 
+    private fun onCarAppStateUpdate(carAppState: CarAppState) {
+        val navigationView = navigationView ?: return
+        when (carAppState) {
+            FreeDriveState -> {
+                logI(LOG_TAG, "navigationView.api.startFreeDrive()")
+                navigationView.api.startFreeDrive()
+            }
+            RoutePreviewState -> {
+                logI(LOG_TAG, "navigationView.api.startRoutePreview()")
+                navigationView.api.startRoutePreview()
+            }
+            ActiveGuidanceState -> {
+                logI(LOG_TAG, "navigationView.api.startActiveGuidance()")
+                val routes = MapboxNavigationApp.current()!!.getNavigationRoutes()
+                navigationView.api.startActiveGuidance(routes)
+            }
+            ArrivalState -> {
+                logI(LOG_TAG, "navigationView.api.startArrival()")
+                navigationView.api.startArrival()
```

Comment:
You should pass routes here, like for `ActiveGuidanceState`. Otherwise if you launch the app on a phone, while Android Auto is in `ArrivalState `, Android Auto will automatically transition to `FreeDriveState`, but arrival state should be propagated to the phone instead. That's the only issue I was able to find, great work!

Reply:
Thanks for finding! https://github.com/mapbox/mapbox-navigation-android-examples/pull/152